process/block/shardblock.go

```diff
@@ -378,6 +378,10 @@ func (sp *shardProcessor) checkEpochCorrectness(
 	incorrectStartOfEpochBlock := header.GetEpoch() != currentBlockHeader.GetEpoch() &&
 		sp.epochStartTrigger.MetaEpoch() == currentBlockHeader.GetEpoch()
 	if incorrectStartOfEpochBlock {
+		if header.IsStartOfEpochBlock() {
```

Comment:
I gave an approve. But still thinking if this opens up other problems or not. Cannot say for sure know. But can't confirm it either, yet.